File: Problems 1700-1799/1701. Average Waiting Time/solution.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        # already sorted by arrival time
        currTime = 0
        total = 0
        for arrival, time in customers:
            if currTime < arrival:
                currTime = arrival
            currTime += time
            total += currTime - arrival

        return total / len(customers)

File: Problems 1700-1799/1701. Average Waiting Time/solution.cpp
class Solution {
    public:
        double averageWaitingTime(vector<vector<int>>& customers) {
            int n = customers.size();
            int currTime = 0;
            double waitingTime = 0;
            for (int i = 0; i < n; i++) {
                int arrival = customers[i][0];
                int time = customers[i][1];
                if (currTime < arrival) currTime = arrival;
                currTime += time;
                waitingTime += currTime - arrival;
            }
    
            return 1.0 * waitingTime / n;
        }
    };